File: app/src/main/java/me/ycdev/android/devtools/root/cmd/RootCmdBase.java
package me.ycdev.android.devtools.root.cmd;

import android.content.Context;
import android.text.TextUtils;

import eu.chainfire.libsuperuser.Shell;

public abstract class RootCmdBase extends ContextCmdBase {
    public RootCmdBase(Context cxt) {
        super(cxt);
    }

    public static void runSuCommand(String[] cmds, String selinuxContext) {
        if (!TextUtils.isEmpty(selinuxContext) && isSELinuxEnforced()) {
            String shell = Shell.SU.shell(0, selinuxContext);
            Shell.run(shell, cmds, null, false);
        } else {
            Shell.SU.run(cmds);
        }
    }

    private static boolean isSELinuxEnforced() {
        return Shell.SU.isSELinuxEnforcing();
    }
}

File: app/src/main/java/me/ycdev/android/devtools/root/cmd/RootCmdBase.kt
package me.ycdev.android.devtools.root.cmd

import android.content.Context

// import eu.chainfire.libsuperuser.Shell;
abstract class RootCmdBase(cxt: Context) : ContextCmdBase(cxt) {
    companion object {
        fun runSuCommand(
            cmds: Array<String>,
            selinuxContext: String?
        ) { //        if (!TextUtils.isEmpty(selinuxContext) && isSELinuxEnforced()) {
//            String shell = Shell.SU.shell(0, selinuxContext);
//            Shell.run(shell, cmds, null, false, true);
//        } else {
//            Shell.SU.run(cmds);
//        }
        } //    private static boolean isSELinuxEnforced() {
//        return Shell.SU.isSELinuxEnforcing();
//    }
    }
}
